app/controllers/thread_controller.php

```diff
@@ -62,19 +65,53 @@ public function create()
     {
         $thread = new Thread;
         $comment = new Comment;
-        $page = Param::get('page_next', 'create');
+        $page = Param::get(Thread::PAGE_NEXT, Thread::PAGE_CREATE);
+        $user_id = Session::get('id');
 
         switch ($page) {
-            case 'create':
+            case Thread::PAGE_CREATE:
                 break;
-            case 'create_end':
+            case Thread::PAGE_CREATE_END:
                 $thread->title = Param::get('title');
+                $thread->user_id = $user_id;
                 $comment->username = Param::get('username');
                 $comment->body = Param::get('body');
+
+                $thread->validate();
+                $comment->validate();
+                if($thread->hasError() || $comment->hasError())
+                {
+                    $page = Thread::PAGE_CREATE;  
+                } else {
+                    $thread->create($comment, Session::get('username'), Session::get('id'));
+                }
+                break;
+            default:
+                throw new NotFoundException("{$page} is not found");
+                break;
+        }
+        $this->set(get_defined_vars());
+        $this->render($page);
+    }
+
+    public function edit()
+    {
+        $thread = new Thread;
```

Comment:
similar comment to: https://github.com/j-asis/Web-Portal/pull/2#discussion_r28211420
:bow: